packages/php-wasm/compile/build-assets/phpwasm-emscripten-library.js

```diff
@@ -401,10 +401,21 @@ const LibraryExample = {
 	},
 
 	js_module_onMessage: function (data) {
+		if (typeof Asyncify === 'undefined') {
+			return;
+		}
+
 		if (Module['onMessage']) {
 			const dataStr = UTF8ToString(data);
-			
-			Module['onMessage'](dataStr);
+
+			return Asyncify.handleSleep((wakeUp) => {
+				Module['onMessage'](dataStr).then((result) => {
+					wakeUp(allocateUTF8OnStack(result));
+				}).catch((e) => {
+					console.error(e);
+					wakeUp('error');
```

Comment:
Ah, hm, that's the C function. But what should the PHP function return? Maybe a string on success and `false` on error, to follow the convention from other PHP functions?

Reply:
null seems fine?

> Then it could return success information, like 1 on success and 0 on error, as well as provide any string via **output. What do you think @ellatrix @dmsnell ?

Not sure I follow here.

Reply:
`NULL` doesn't tell you that something in JavaScript triggered an error. Which hm, thinking about this again, I'm not sure PHP should even care. This is just a message passing mechanism, if there's an error the response just won't come. Alright, let's stick with NULL for now – the `wakeUp('error')` call will still need adjusting as `"error"` is not a valid pointer address.

Reply:
I updated the PR to do that